cloud/services/loadbalancers.go

```diff
@@ -70,6 +86,48 @@ func EnsureNodeBalancer(ctx context.Context, clusterScope *scope.ClusterScope, l
 	return clusterScope.LinodeClient.CreateNodeBalancer(ctx, createConfig)
 }
 
+// getSubnetID returns the subnetID of the first subnet in the LinodeVPC.
+// If no subnets or subnetID is found, it returns an error.
+func getSubnetID(ctx context.Context, clusterScope *scope.ClusterScope, logger logr.Logger) (int, error) {
+	name := clusterScope.LinodeCluster.Spec.VPCRef.Name
+	namespace := clusterScope.LinodeCluster.Spec.VPCRef.Namespace
+	if namespace == "" {
+		namespace = clusterScope.LinodeCluster.Namespace
+	}
+
+	logger = logger.WithValues("vpcName", name, "vpcNamespace", namespace)
+
+	linodeVPC := &v1alpha2.LinodeVPC{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+	}
+
+	objectKey := client.ObjectKeyFromObject(linodeVPC)
+	err := clusterScope.Client.Get(ctx, objectKey, linodeVPC)
+	if err != nil {
+		logger.Error(err, "Failed to fetch LinodeVPC")
+		return -1, err
+	}
+	if len(linodeVPC.Spec.Subnets) == 0 {
+		err = errors.New("No subnets found in LinodeVPC")
+		logger.Error(err, "Failed to fetch LinodeVPC")
+		return -1, err
+	}
+
+	// We are using the first subnet as the default one
+	// get the subnetID from the linodeVPC object
+	subnetID := linodeVPC.Spec.Subnets[0].SubnetID
```

Comment:
If `clusterScope.LinodeCluster.Spec.Network.SubnetName` is specified, we should use its id as subnetID. A single VPC can have multiple clusters which means there can be more than one subnets and we might want to use different subnet than the first subnet.